huaweicloud/services/apig/resource_huaweicloud_apig_group.go

```diff
@@ -315,6 +366,77 @@ func updateEnvironmentVariables(client *golangsdk.ServiceClient, d *schema.Resou
 	return createEnvironmentVariables(client, instanceId, groupId, addRaws)
 }
 
+func associateDomain(client *golangsdk.ServiceClient, domains []interface{}, instanceId string, groupId string) error {
+	for _, v := range domains {
+		domain := v.(map[string]interface{})
+		opts := apigroups.AssociateDomainOpts{
+			InstanceId:            instanceId,
+			GroupId:               groupId,
+			UrlDomain:             domain["name"].(string),
+			MinSSLVersion:         domain["min_ssl_version"].(string),
+			IsHttpRedirectToHttps: domain["is_http_redirect_to_https"].(bool),
+		}
+		_, err := apigroups.AssociateDomain(client, opts)
+		if err != nil {
+			return fmt.Errorf("error binding domain name: %s", err)
+		}
+	}
+	return nil
+}
+
+func getDomainId(client *golangsdk.ServiceClient, name string, instanceId string, groupId string) (string, error) {
+	resp, err := apigroups.Get(client, instanceId, groupId).Extract()
+	if err != nil {
+		return "", fmt.Errorf("error retrieving dedicated group: %s", err)
+	}
+
+	if len(resp.UrlDomians) == 0 {
+		return "", fmt.Errorf("unable to find any domain name information under dedicated group:%s", groupId)
+	}
+
+	for _, v := range resp.UrlDomians {
+		if v.DomainName == name {
+			return v.Id, nil
+		}
+	}
+	return "", nil
+}
+
+func disAssociateDomain(client *golangsdk.ServiceClient, domains []interface{}, instanceId string, groupId string) error {
+	for _, v := range domains {
+		domain := v.(map[string]interface{})
+		domainId, err := getDomainId(client, domain["name"].(string), instanceId, groupId)
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		domainName := domain["name"].(string)
		domainId, err := getDomainId(client, instanceId, groupId, domainName)
		if err != nil {
			if _, ok := err.(golangsdk.ErrDefault404); ok {
				log.Printf("[DEBUG] The domain (%s) has been disassociated.", domainName)
				continue
			}
			return err
		}
```
If the ID does not exist, an error is reported when the API is invoked.

Reply:
fixed